From 44706a6dce64b0e9289769dcf2162aad9dbb3973 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Mon, 8 Jun 2020 06:18:02 +0530
Subject: [PATCH 09/25] gutils: GetWindowsDirectory is not available on WinRT

This call is only allowed on desktop apps, not UWP apps:

https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getwindowsdirectorya#requirements
---
 glib/gutils.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/glib/gutils.c b/glib/gutils.c
index cc125e1..81eea9f 100644
--- a/glib/gutils.c
+++ b/glib/gutils.c
@@ -342,9 +342,11 @@ g_find_program_in_path (const gchar *program)
   if (n > 0 && n < MAXPATHLEN)
     sysdir = g_utf16_to_utf8 (wsysdir, -1, NULL, NULL, NULL);
   
+#ifndef G_WINAPI_ONLY_APP
   n = GetWindowsDirectoryW (wwindir, MAXPATHLEN);
   if (n > 0 && n < MAXPATHLEN)
     windir = g_utf16_to_utf8 (wwindir, -1, NULL, NULL, NULL);
+#endif
   
   if (filename)
     {
@@ -549,6 +551,7 @@ get_windows_directory_root (void)
 {
   wchar_t wwindowsdir[MAX_PATH];
 
+#ifndef G_WINAPI_ONLY_APP
   if (GetWindowsDirectoryW (wwindowsdir, G_N_ELEMENTS (wwindowsdir)))
     {
       /* Usually X:\Windows, but in terminal server environments
@@ -567,6 +570,7 @@ get_windows_directory_root (void)
       return windowsdir;
     }
   else
+#endif
     return g_strdup ("C:\\");
 }
 
-- 
2.38.1

